src/dropdown/dropdown.js

```diff
@@ -187,17 +187,31 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
     }
   };
 
+  function toggleDropdownUp() {
+    var clientRect = $element[0].getBoundingClientRect(),
+      dropdownHeight = scope.getDropdownElement().outerHeight();
+
+    if (clientRect.top > dropdownHeight &&
+        window.innerHeight - clientRect.top < clientRect.height + dropdownHeight) {
+      return true;
+    }
+    return false;
+  }
+
   scope.$watch('isOpen', function(isOpen, wasOpen) {
     if (appendTo && self.dropdownMenu) {
       var pos = $position.positionElements($element, self.dropdownMenu, 'bottom-left', true),
```

Comment:
The placement parameter passed to the positionElements method should be changed to 'auto bottom-left' and leverage the work already being done there.